### DISCUSSION TITLE
image in public folder giving 404

### DISCUSSION THREAD

**Discussion by dagda1** [remix-run/remix#6090] [Q&A]
I have a file in the following location

`/public/images/displacement.jpg` 

but when I try and serve the image at `http://localhost:3333/images/displacement.jpg` I can see in the console:

> No routes matched location "/images/displacement.jpg"

And it gives 404.

I tried adding this preload to `links` but no cigar

```ts
{
  rel: 'preload',
  href: '/images/displacement.jpg',
  as: 'image',
},
```

**Comment by dagda1** [comment]
I figured it out.  I just import from `/app` like a regular import

```ts
import displacement from '~/images/displacement.jpg';
```

and the runtime adds the image to 

> ~/_static/build/_assets/

**Reply by kiliman** [reply]
You shouldn't have to do that. What server are you running? Remix App Server/Express? They should automatically serve any files in your public folder before it even gets to Remix (static file handler).

You typically import assets if you know they'll change, and you want to ensure the file hash changes and busts the cache.
